src/libs/utils.ts

```diff
@@ -41,16 +41,21 @@ export interface Atom<T> {
   reset: () => void;
 }
 
+type AtomFactory = {
+  <T = unknown>(initialValue: T): Atom<T>;
+  <T = unknown>(): Atom<T | undefined>;
```

Comment:
Hmmm... I am pretty sure that we should force the explicit (allow undefined) onto the individual instances that define T.  Allowing undefined here as a blanket seems problematic.  For example `Atom<MyType>`  where `type MyType = FancyThing[] | undefined`

Reply:
If an initial value is provided, undefined is _not_ included in the type.

For example:
![Screenshot 2023-08-24 at 9 56 46 AM](https://github.com/DataBiosphere/terra-ui/assets/1156625/1b57baa6-a9d6-44d1-8a2a-8f2963b5f0f7)

Reply:
Updated to require an initial value in https://github.com/DataBiosphere/terra-ui/pull/4148/commits/a2a8bc931d44d00a5cbf4a3a52edb579122b5341.